steward/src/commands/keys/eth/import_from_mnemonic.rs

```diff
@@ -44,33 +44,20 @@ impl Runnable for ImportEthKeyCmd {
                 .expect("Could not read mnemonic"),
         };
 
-        let key = match bip32::Mnemonic::new(mnemonic.trim(), Default::default()) {
-            Ok(mnemonic) => {
-                let seed = mnemonic.to_seed("");
+        if let Ok(mnemonic) = bip32::Mnemonic::new(mnemonic.trim(), Default::default()) {
+            let seed = mnemonic.to_seed("");
+            let path = config.ethereum.key_derivation_path.trim();
+            let path = path
+                .parse::<bip32::DerivationPath>()
+                .expect("Could not parse derivation path");
 
-                let path = config.ethereum.key_derivation_path.trim();
-                let path = path
-                    .parse::<bip32::DerivationPath>()
-                    .expect("Could not parse derivation path");
-
-                let key = bip32::XPrv::derive_from_path(seed, &path).expect("Could not derive key");
-                let key = k256::SecretKey::from(key.private_key());
-                key.to_pkcs8_der()
-                    .expect("Could not PKCS8 encod private key")
-            }
-            Err(_) => {
-                let key = rpassword::read_password_from_tty(Some("> Enter your private-key:\n"))
-                    .expect("Could not read private-key");
-
-                let key: ethers::types::H256 = key.parse().expect("Could not parse private-key");
-
-                let key = k256::SecretKey::from_bytes(key).expect("Could not make private key");
-                key.to_pkcs8_der()
-                    .expect("Could not PKCS8 encode private key")
-            }
-        };
-
-        keystore.store(&name, &key).expect("Could not store key");
+            let key = bip32::XPrv::derive_from_path(seed, &path).expect("Could not derive key");
+            let key = k256::SecretKey::from(key.private_key());
+            let key = key
+                .to_pkcs8_der()
+                .expect("Could not PKCS8 encod private key");
+            keystore.store(&name, &key).expect("Could not store key");
+        }
```

Comment:
We should error out if we don't get the `Ok` back from this.

Reply:
Since it won't be possible to return the error to the user, is it worth it? For instance, from the docs and other places I've seen this used, I'll have to use a panic, println, error, or other macros, and return a string.

https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html

I used the error macro for this though, let me know what you think.